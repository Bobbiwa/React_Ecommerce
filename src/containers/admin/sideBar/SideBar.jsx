import React from 'react';
import { Menu } from 'antd';
import { useNavigate } from 'react-router-dom'
import menuList from '../../../config/menuConfig'
import logo from '../../../static/imgs/EpamLogo.jpeg'
import './css/sidebar.less'


//Functions of antD
function getItem(label, key, icon, children, type) {
    return {
        key,
        icon,
        children,
        label,
        type,
    };
}

//Own function: Making a list by recursion
const getList = (menuList) => {
    return menuList.reduce((pre, cur) => {
        if (!cur.children) {
            let resItem = getItem(cur.title, cur.key, cur.icon)
            return [...pre, resItem]
        } else {
            let resItem = getItem(cur.title, cur.key, cur.icon,
                getList(cur.children)
            )
            return [...pre, resItem]
        }
    }, [])
}
const items = getList(menuList)  //[{...},{...}]


export default function SideBar() {
    const navigate = useNavigate()
    const {pathname} = window.location


    //Click to jump
    const link = (e) => {
        navigate(e.key)
    }
    return (
        <div className='side_bar'>
            <img src={logo} className='logo_img' alt="Epam's logo" />
            <Menu
                className='menu'
                defaultSelectedKeys = {pathname.indexOf('product') !== -1? '/admin/product-about/product':pathname }
                defaultOpenKeys = {pathname.split('/').splice(2)}
                mode="inline"
                theme="dark"
                items={items}
                onClick={link}
            />
        </div>
    )
}
