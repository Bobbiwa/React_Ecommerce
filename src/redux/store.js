// import {legacy_createStore as createStore,applyMiddleware} from 'redux'
// import reducers from './reducers'
// import {composeWithDevTools} from 'redux-devtools-extension'
// import thunk from 'redux-thunk'

// export default createStore(reducers,composeWithDevTools(applyMiddleware(thunk)))

import { configureStore } from '@reduxjs/toolkit'
import userInfoSlice from './slice/userInfoSlice'
import categorySlice from './slice/categorySlice'
import productSlice from './slice/productSlice'
export default configureStore({
    reducer: {
        userInfo: userInfoSlice,
        categoryInfo: categorySlice,
        productInfo: productSlice
    }
})