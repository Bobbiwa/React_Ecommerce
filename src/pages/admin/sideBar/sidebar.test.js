import React from "react";
import { cleanup, render, screen } from '@testing-library/react'
import {BrowserRouter} from 'react-router-dom'
import {Provider} from 'react-redux'
import store from '../../../redux/store'
import SideBar from './SideBar'

afterEach(cleanup)
it('should take a snapshot',()=>{
    const {asFragment} = render(<Provider store={store}><BrowserRouter><SideBar /></BrowserRouter></Provider>)
    expect(asFragment()).toMatchSnapshot()
})
it('should equal to Admin', () => {
    render(<Provider store={store}><BrowserRouter><SideBar /></BrowserRouter></Provider>)
    expect(screen.getByText('Admin')).toHaveTextContent('Admin')
})
it('should equal to Product', () => {
    render(<Provider store={store}><BrowserRouter><SideBar /></BrowserRouter></Provider>)
    expect(screen.getByText('Product')).toHaveTextContent('Product')
})
it('should equal to User Management', () => {
    render(<Provider store={store}><BrowserRouter><SideBar /></BrowserRouter></Provider>)
    expect(screen.getByText('User Management')).toHaveTextContent('User Management')
})
it('should equal to Role Management', () => {
    render(<Provider store={store}><BrowserRouter><SideBar /></BrowserRouter></Provider>)
    expect(screen.getByText('Role Management')).toHaveTextContent('Role Management')
})
it('should equal to Charts', () => {
    render(<Provider store={store}><BrowserRouter><SideBar /></BrowserRouter></Provider>)
    expect(screen.getByText('Charts')).toHaveTextContent('Charts')
})