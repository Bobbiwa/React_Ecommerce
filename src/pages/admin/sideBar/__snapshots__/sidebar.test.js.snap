// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="side_bar"
  >
    <img
      alt="Epam's logo"
      class="logo_img"
      src="EpamLogo.jpeg"
    />
    <ul
      class="ant-menu ant-menu-root ant-menu-inline ant-menu-dark menu"
      data-menu-list="true"
      role="menu"
      tabindex="0"
    >
      <li
        class="ant-menu-item"
        data-menu-id="rc-menu-uuid-test-/admin/home"
        role="menuitem"
        style="padding-left: 24px;"
        tabindex="-1"
      >
        <span
          aria-label="home"
          class="anticon anticon-home ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="home"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
            />
          </svg>
        </span>
        <span
          class="ant-menu-title-content"
        >
          Admin
        </span>
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-inline"
        role="none"
      >
        <div
          aria-controls="rc-menu-uuid-test-product-about-popup"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          data-menu-id="rc-menu-uuid-test-product-about"
          role="menuitem"
          style="padding-left: 24px;"
          tabindex="-1"
        >
          <span
            aria-label="appstore"
            class="anticon anticon-appstore ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="appstore"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M464 144H160c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V160c0-8.8-7.2-16-16-16zm-52 268H212V212h200v200zm452-268H560c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V160c0-8.8-7.2-16-16-16zm-52 268H612V212h200v200zM464 544H160c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V560c0-8.8-7.2-16-16-16zm-52 268H212V612h200v200zm452-268H560c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V560c0-8.8-7.2-16-16-16zm-52 268H612V612h200v200z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            Product
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item"
        data-menu-id="rc-menu-uuid-test-/admin/user"
        role="menuitem"
        style="padding-left: 24px;"
        tabindex="-1"
      >
        <span
          aria-label="user"
          class="anticon anticon-user ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="user"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
            />
          </svg>
        </span>
        <span
          class="ant-menu-title-content"
        >
          User Management
        </span>
      </li>
      <li
        class="ant-menu-item"
        data-menu-id="rc-menu-uuid-test-/admin/role"
        role="menuitem"
        style="padding-left: 24px;"
        tabindex="-1"
      >
        <span
          aria-label="edit"
          class="anticon anticon-edit ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="edit"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
            />
          </svg>
        </span>
        <span
          class="ant-menu-title-content"
        >
          Role Management
        </span>
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-inline"
        role="none"
      >
        <div
          aria-controls="rc-menu-uuid-test-charts-popup"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          data-menu-id="rc-menu-uuid-test-charts"
          role="menuitem"
          style="padding-left: 24px;"
          tabindex="-1"
        >
          <span
            aria-label="area-chart"
            class="anticon anticon-area-chart ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="area-chart"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M888 792H200V168c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v688c0 4.4 3.6 8 8 8h752c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm-616-64h536c4.4 0 8-3.6 8-8V284c0-7.2-8.7-10.7-13.7-5.7L592 488.6l-125.4-124a8.03 8.03 0 00-11.3 0l-189 189.6a7.87 7.87 0 00-2.3 5.6V720c0 4.4 3.6 8 8 8z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            Charts
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
    </ul>
    <div
      aria-hidden="true"
      style="display: none;"
    />
  </div>
</DocumentFragment>
`;
